src/server/api/users.js

```diff
@@ -1,10 +1,77 @@
 const express = require("express");
 const router = express.Router();
-// const prisma = require("../client");
+const { getUser, getAllUsers, updateUser, deleteUser } = require("../../../prisma/users");
+
+const prisma = require("../../../prisma/client");
+
+// ROUTE TO CREATE USERS IS IN server/auth/register
 
 // GET /api/users
 router.get("/", async (req, res, next) => {
-  res.send("Hello user route!");
+  try {
+    const users = await getAllUsers();
+    res.status(200).send(users);
+  } catch (error) {
+    console.error(error);
+  } finally {
+    await prisma.$disconnect();
+  }
+});
+
+//GET /api/users/:id
+router.get("/:id", async (req, res, next) => {
+  const userId = parseInt(req.params.id);
+
+  try {
+    const user = await getUser(userId);
+    if (!user) {
+      return res.status(404).send("User not found!");
+    }
+    res.send(user);
+  } catch (error) {
+    console.error("Error retrieving user by ID:", error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+//PUT /api/users/:id
+router.put("/:id", async (req, res, next) => {
```

Comment:
router.patch, not router.put. According to ChatGPT on the differences, with a PUT request, "if you omit a field in your update, it might be overwritten with a null or default value on the server." With a PATCH request, "When you send a PATCH request, you typically send only the fields that need to be updated. The server then applies those changes to the existing resource."